src/components/settings/SettingsLMSTab/LMSConfigs/SAPConfig.jsx

```diff
@@ -82,17 +82,24 @@ const SAPConfig = ({ enterpriseCustomerUuid, onClick, existingData }) => {
     switch (field) {
       case 'SAP Base URL':
         setSapsfBaseUrl(input);
-        setUrlValid(urlValidation(input) || input.length === 0);
+        setUrlValid(urlValidation(input) || input?.length === 0);
         break;
       case 'Display Name':
         setDisplayName(input);
-        setNameValid(input.length <= 20);
+        setNameValid(input?.length <= 20);
         break;
       default:
         break;
     }
   };
 
+  useEffect(() => {
```

Comment:
Due to the multiple impl we have for all the LMS configs, I see a fewopportunities for DRY; stuff like this could be a common hook for example, all the implementations use it as a validation hook of sorts (can pass in custom validation functions for example), but I don't think it's needed right now, just mentioning as a future idea if the codebase becomes larger

this could e.g., become a nice validation hook all components can use or be required to us-e by default 

the hook would take some inputs like field names, and take care of the actual validating via other shared `validateField` functions, but it would require refactors :)